src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -93,11 +97,22 @@ export class ElasticsearchService
     };
   }
   public async start(): Promise<InternalElasticsearchServiceStart> {
-    if (!this.client) {
+    if (!this.client || !this.esNodesCompatibility$) {
       throw new Error('ElasticsearchService needs to be setup before calling start');
     }
 
     const config = await this.config$.pipe(first()).toPromise();
+
+    // Log every error we may encounter in the connection to Elasticsearch
+    this.esNodesCompatibility$.subscribe(({ isCompatible, message }) => {
+      if (!isCompatible && message) {
```

Comment:
Looking at the logic, when `isCompatible: false`, `message` always exist. Only when everything is alright, it doesn't return any `message`. 
I think that this is a case where we don't want to be super-explicit with TS for simplicity. Functional-wise we may want to add the default message just in case, but it would never make it to coverage.

I've created #109654 to discuss it further :)